Add metrics and resource attribute config options from vcentereceiver in docs and tests (#5813)
grafana/agent: docs/sources/flow/reference/components/otelcol.receiver.vcenter.md

Hierarchy | Block | Description | Required
--------- | ----- | ----------- | --------
tls | [tls][] | Configures TLS for the HTTP client. | no

> metrics | [metrics][] | Configures the metrics that this component generates to monitor its state. | no
> resource_attributes | [resource_attributes][] | Configures the resource attributes that this component generates to annotate telemetry data. | no

Reviewer: ```suggestion
metrics | [metrics][] | Configures which metrics will be sent to downstream components. | no
resource_attributes | [resource_attributes][] | Configures resource attributes for metrics sent to downstream components. | no
```
`metrics` is not for the component's state :) It's the state of the vcenter endpoint.

{{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" version="<AGENT VERSION>" >}}

### metrics block

> Name | Type | Description | Default | Required

Reviewer: Do you mind making `metric` in the `Type` column a hyperlink to `#### metric block` please?
BTW After we get consensus on #5764, I might need to rework this doc to be in line with that future desicion.
